analysis/table1.R

```diff
@@ -0,0 +1,216 @@
+  colnames(df)[grepl("strat_cat_", colnames(df))],
+  colnames(df)[grepl("cov_bin_", colnames(df))]
+)]
+
+df$All <- "All"
+
+# Filter binary data
+
+for (colname in colnames(df)[grepl("cov_bin_", colnames(df))]) {
+  df[[colname]] <- sapply(df[[colname]], as.character)
+}
+
+df <- df %>%
+  mutate(across(where(is.factor), as.character))
+
+# Aggregate data ---------------------------------------------------------------
+print("Aggregate data")
+
+df <- tidyr::pivot_longer(
+  df,
+  cols = setdiff(colnames(df), c("patient_id", "exposed")),
+  names_to = "characteristic",
+  values_to = "subcharacteristic"
+)
+#as.factor or factor to convert binary functions
+
+df$total <- 1
+
+df <- aggregate(
+  cbind(total, exposed) ~ characteristic + subcharacteristic,
+  data = df,
+  sum
+)
+
+# Tidy care home characteristic ------------------------------------------------
+print("Remove extraneous information")
+
+df <- df[df$subcharacteristic != FALSE, ]
+df$subcharacteristic <- ifelse(
+  df$subcharacteristic == "" | df$subcharacteristic == "unknown",
+  "Missing",
+  df$subcharacteristic
+)
+
+# Sort characteristics ---------------------------------------------------------
+print("Sort characteristics")
+
+df <- df[order(df$subcharacteristic, decreasing = TRUE), ] # Basic sort from extendedtable, might need to clean up
+df <- df[order(df$characteristic), ]
+
+# Add in Median IQR
+print('Add median (IQR) age')
+
+# Pastes: "Mean Age (LQ Age - UQ Age)" as a string for each cohort
+df[nrow(df) + 1, ] <- c("Age, years", "Median (IQR)", median_iqr_string, 0)
+
+# Define table1 output folder ---------------------------------------------------------
+print("Creating output/table1 output folder")
+
+# setting up the sub directory
+table1_dir <- "output/table1/"
+
+# check if sub directory exists, create if not
+fs::dir_create(here::here(table1_dir))
+
+# Save Table 1 -----------------------------------------------------------------
+print("Save Table 1")
+
+write.csv(df, paste0(table1_dir, "table1_", cohort, ".csv"), row.names = FALSE)
+
+# Perform redaction ------------------------------------------------------------
+print("Perform redaction")
+
+df <- df[df$subcharacteristic != "Median (IQR)", ] # Remove Median IQR row
+
+df$total_midpoint6 <- roundmid_any(as.numeric(df$total), to = threshold)
+df$exposed_midpoint6 <- roundmid_any(as.numeric(df$exposed), to = threshold)
+
+# Calculate column percentages -------------------------------------------------
+
+df$N_midpoint6_derived <- df$total
+
+df$percent_midpoint6_derived <- paste0(
+  ifelse(
+    df$characteristic == "All",
+    "",
+    paste0(
+      " (",
+      round(
+        100 *
+          (df$total_midpoint6 /
+            df[df$characteristic == "All", "total_midpoint6"]),
+        1
+      ),
+      "%)"
+    )
+  )
+)
+
+df <- df[, c(
+  "characteristic",
+  "subcharacteristic",
+  "N_midpoint6_derived",
+  "percent_midpoint6_derived",
+  "exposed_midpoint6"
+)]
+
+colnames(df) <- c(
```

Comment:
Better to code this explicity using `dplyr::rename()` so you know which columns are renamed as what.